api/src/auth.js

```diff
@@ -40,53 +28,12 @@ const hasPermission = (userCtx, permission) => {
   return _.some(roles, role => _.includes(userCtx.roles, role));
 };
 
-const hydrateUserSettings = (userSettings) => {
-  return db.medic
-    .allDocs({ keys: [ userSettings.facility_id, userSettings.contact_id ], include_docs: true })
-    .then((response) => {
-      if (!Array.isArray(response.rows) || response.rows.length !== 2) { // malformed response
-        return userSettings;
-      }
-
-      const [facilityRow, contactRow] = response.rows;
-      if (!facilityRow || !contactRow) { // malformed response
-        return userSettings;
-      }
-
-      userSettings.facility = facilityRow.doc;
-      userSettings.contact = contactRow.doc;
-
-      return userSettings;
-    });
-};
-
 module.exports = {
-  hasOnlineRole: roles => {
-    if (!Array.isArray(roles) || !roles.length) {
-      return false;
-    }
-
-    const onlineRoles = [
-      DB_ADMIN_ROLE,
-      ONLINE_ROLE,
-      'national_admin', // kept for backwards compatibility
-    ];
-    return roles.some(role => onlineRoles.includes(role));
-  },
-
-  isOnlineOnly: userCtx => {
-    return userCtx && module.exports.hasOnlineRole(userCtx.roles);
-  },
-
-  isOffline: roles => {
-    const configured = config.get('roles') || {};
-    const configuredRole = roles.some(role => configured[role]);
-    return !isDbAdmin({ roles }) &&
-           (!configuredRole || roles.some(role => configured[role] && configured[role].offline));
-  },
-
+  isOnlineOnly: roles.isOnlineOnly,
+  isDbAdmin: roles.isDbAdmin,
+  getUserSettings: users.getUserSettings,
```

Comment:
I ended up keeping these three references here as pass-throughs because these exported functions were being used in a number of other places.  It feels a little janky, but the alternative is to further expand the number of files being touched in this refactor...